internal/convenience/services/service_test.go

```diff
@@ -255,3 +264,44 @@ func (s *ConvenienceServiceSuite) TestCreateReportIdempotency() {
 	s.NoError(err)
 	s.Equal(1, int(count))
 }
+
+func (s *ConvenienceServiceSuite) TestCreateInputIdempotency() {
+	ctx := context.Background()
+	_, err := s.service.CreateInput(ctx, &model.AdvanceInput{
+		Index:       1,
+		AppContract: common.HexToAddress("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+	})
+	s.NoError(err)
+	count, err := s.inputRepository.Count(ctx, nil)
+	s.NoError(err)
+	s.Equal(1, int(count))
+
+	_, err = s.service.CreateInput(ctx, &model.AdvanceInput{
+		Index:       1,
+		AppContract: common.HexToAddress("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+	})
+	s.NoError(err)
+	count, err = s.inputRepository.Count(ctx, nil)
+	s.NoError(err)
+	s.Equal(1, int(count))
+}
+
+func (s *ConvenienceServiceSuite) TestCreateInputIdempotencyWithoutAppContract() {
+	ctx := context.Background()
+	_, err := s.service.CreateInput(ctx, &model.AdvanceInput{
+		Index: 1,
+	})
+	s.NoError(err)
+	count, err := s.inputRepository.Count(ctx, nil)
+	s.NoError(err)
+	s.Equal(1, int(count))
+
+	_, err = s.service.CreateInput(ctx, &model.AdvanceInput{
+		Index:       1,
+		AppContract: common.HexToAddress("0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"),
+	})
+	s.NoError(err)
+	count, err = s.inputRepository.Count(ctx, nil)
+	s.NoError(err)
+	s.Equal(1, int(count))
```

Comment:
Why is it still 1?

Reply:
The test is wrong, and there is something that doesn't make sense anymore, on Create from input_repository, which is FindByIndex